### PR TITLE
[SPARK-40948][SQL][FOLLOWUP] Restore PATH_NOT_FOUND

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

The original PR to introduce the error class `PATH_NOT_FOUND` was reverted since it breaks the tests in different test env.

This PR proposes to restore it back.

### Why are the changes needed?

Restoring the reverted changes with proper fix.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

The existing CI should pass.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala [hunk 1]
```diff
@@ -2330,7 +2330,7 @@ class DataFrameSuite extends QueryTest
           new File(uuid, "file3").getAbsolutePath,
           uuid).rdd
       }
-      assert(e.getMessage.startsWith("Path does not exist"))
+      assert(e.getErrorClass == "PATH_NOT_FOUND")
```
Comment (itholic) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala:2333]: The current `checkError` seems not allow pattern matching for `parameter` checking.

Let me just check the errorClass for now.

Comment (LuciferYang) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala:2333]: nit: Can the empty `finally` block in this case be cleaned together in this pr?

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala:2333]: > The current checkError seems not allow pattern matching for parameter checking.

It does, see https://github.com/apache/spark/blob/74668e2bf14760dbc60509f7736f410c09084697/core/src/test/scala/org/apache/spark/SparkFunSuite.scala#L310

Please, use `checkError` here. Just set `matchPVals` to `true`.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by LuciferYang** [comment]
Looks like we need to refactor this case

https://github.com/apache/spark/blob/c5d27603f29437f1686cac70727594c19410a273/R/pkg/tests/fulltests/test_sparkSQL.R#L3986-L3998

**Review by MaxGekk — APPROVED** [review]

File: R/pkg/tests/fulltests/test_sparkSQL.R [hunk 2]
```diff
@@ -3990,12 +3990,16 @@ test_that("Call DataFrameWriter.load() API in Java without path and check argume
   expect_error(read.df(source = "json"),
                paste("Error in load : analysis error - Unable to infer schema for JSON.",
                      "It must be specified manually"))
-  expect_error(read.df("arbitrary_path"), "Error in load : analysis error - Path does not exist")
-  expect_error(read.json("arbitrary_path"), "Error in json : analysis error - Path does not exist")
-  expect_error(read.text("arbitrary_path"), "Error in text : analysis error - Path does not exist")
-  expect_error(read.orc("arbitrary_path"), "Error in orc : analysis error - Path does not exist")
+  expect_error(read.df("arbitrary_path"),
+               "Error in load : analysis error - \\[PATH_NOT_FOUND\\].*")
```
Comment (MaxGekk) [inline R/pkg/tests/fulltests/test_sparkSQL.R:3994]: @HyukjinKwon @srielau @cloud-fan Are you ok with such changes?

Comment (itholic) [reply R/pkg/tests/fulltests/test_sparkSQL.R:3994]: FYI: the `arbitrary_path` here is different per test environments, so I use regexp for the path string.

Comment (cloud-fan) [reply R/pkg/tests/fulltests/test_sparkSQL.R:3994]: LGTM

Comment (HyukjinKwon) [reply R/pkg/tests/fulltests/test_sparkSQL.R:3994]: thanks man

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic and @HyukjinKwon @cloud-fan @LuciferYang for review.

**Pull request closed without merging**